django/db/backends/postgresql/operations.py

```diff
@@ -267,11 +267,10 @@ def distinct_sql(self, fields, params):
             return ["DISTINCT"], []
 
     def last_executed_query(self, cursor, sql, params):
-        # https://www.psycopg.org/docs/cursor.html#cursor.query
-        # The query attribute is a Psycopg extension to the DB API 2.0.
-        if cursor.query is not None:
-            return cursor.query.decode()
-        return None
+        try:
+            return compose(sql, params, cursor.connection)
```

Comment:
This might also the origin of some of the slowdown but it should only affect testing. It'd be great to keep using `cursor.query.decode()` on pyscopg2 though to avoid any regressions.